terraform/transform_provider.go

```diff
@@ -113,39 +120,17 @@ func (t *ProviderTransformer) Transform(g *Graph) error {
 
 			providerAddr, exact := pv.ProvidedBy()
 			var absPc addrs.AbsProviderConfig
-			var providerFqn addrs.Provider
 
 			switch p := providerAddr.(type) {
 			case addrs.AbsProviderConfig:
 				absPc = p
-				// ProvidedBy() returns an AbsProviderConfig + exact == true
-				// when the provider configuration is set in state, so we do not
-				// need to verify the FQN matches.
+				// ProvidedBy() returns an AbsProviderConfig when the provider
+				// configuration is set in state, so we do not need to verify
+				// the FQN matches.
 				if exact {
```

Comment:
Do we still need the `if exact` check in this case? I might be misreading, but it looks to me like `exact` is always true when we return an `AbsProviderConfig`.

Reply:
It is still necessary (though I had to run the tests to be reminded of this): the `graphNodeImportState` node can return an `AbsProviderConfig` + false